hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocol/OzoneManagerProtocol.java

```diff
@@ -82,15 +85,19 @@
    *         already the owner.
    * @throws IOException
    */
-  boolean setOwner(String volume, String owner) throws IOException;
+  default boolean setOwner(String volume, String owner) throws IOException {
+    throw new NotImplementedException("setOwner is not implemented");
```

Comment:
Instead of changing the interface, I think it would be safer to throw exception only in `OzoneManager`'s "implementation".  That way new implementors would be required to implement these methods or deliberately decide not to.

If you agree, then please also consider changing to `UnsupportedOperationException` (as
> [`NotImplementedException`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/NotImplementedException.html) represents the case where the author has yet to implement the logic at this point in the program

which would not be the case with `OzoneManager`.

In the long run I think we should split `OzoneManagerProtocol` interface into separate read and write interfaces.  `OzoneManager` should only implement read interface, and `OzoneManagerProtocolClientSideTranslatorPB` should implement both.

Reply:
> In the long run I think we should split `OzoneManagerProtocol` interface into separate read and write interfaces. `OzoneManager` should only implement read interface, and `OzoneManagerProtocolClientSideTranslatorPB` should implement both.

Thanks @adoroszlai for the thought.
I'm fine with this proposal, and I would split the `OzoneManagerProtocol` interface into separate read and write interfaces in this patch. (Feel free to give me feedback or share your thoughts, thanks.)

Reply:
> In the long run I think we should split `OzoneManagerProtocol` interface into separate read and write interfaces. `OzoneManager` should only implement read interface, and `OzoneManagerProtocolClientSideTranslatorPB` should implement both.

[Edit Comment]
Thanks @adoroszlai for the idea.
I would share some thoughts about this work.